### PR TITLE
[flash_ctrl] Various documentation updates

### LINKED ISSUE #9205: [flash_ctrl] Test plan review action items
- [ ] re-name diagram to flash protocol controller and flash physical controller
- [ ] Add information about how many times the protocol controller can lose arbitration before it escalates
- [ ] Add information that directed host path can only read and can only access data partition
- [ ] Show the jtag path in block level diagram
- [ ] Show all tlul connections in the block level diagram
- [ ] remove metadata fifo path in the diagram
- [ ] Double check closed source testplan and usage of "high endurance" attribute
- [ ] Add description in the memory protection area that clarifies an "overflow" address will automatically be flagged as an error. 
- [ ] update `number of pages per bank` to `number of data partitions pages per bank`
- [ ] add more / better description to the isolated and secret partitions.
- [ ] add examples on how to calculate the address for a particular bank / page

### PR DESCRIPTION
**Pull request by tjaychen**
- fixes #9205

Signed-off-by: Timothy Chen <timothytim@google.com>

### CODE REVIEW

**Comment by tjaychen** [comment]
@a-will
@nikola-miladinovic

File: hw/ip/flash_ctrl/doc/_index.md [hunk 1]
```diff
@@ -250,6 +310,19 @@ The vendor wrapper module is also responsible for any BIST, redundancy handling,
 
 The scramble keys are provided by an external static block such as the OTP.
 
+### Host and Protocol Controller Handling
+
+Both the protocol controller and the system host converge on the physical controller.
+The protocol controller has read access to all partitions as well as programd and erase privileges.
```
Comment (matutem) [inline hw/ip/flash_ctrl/doc/_index.md:316]:
```suggestion
The protocol controller has read access to all partitions as well as program and erase privileges.
```

File: hw/ip/flash_ctrl/doc/_index.md [hunk 2]
```diff
@@ -108,15 +108,69 @@ Lastly, while the different partitions may be identical in some attributes, they
...
+For example:
+Assume each bank is 512KB in size.
+The address of bank 0 is `0x0`.
+The address of bank 1 is `0x80000`
+
+##### Page Address
+The address of a particular page is calculated based on the page size and the index number of the page.
+
+For example:
+Assume each page is 2KB in size.
```
Comment (matutem) [inline hw/ip/flash_ctrl/doc/_index.md:128]: But line 59 suggests 1KB pages... Could you clarify?

**Review by matutem — APPROVED** [review]
Please address the comments, and then LGTM.

File: hw/ip/flash_ctrl/doc/_index.md (hunk 2 above)
Comment (tjaychen) [reply hw/ip/flash_ctrl/doc/_index.md:128]: gasp! did i do my math wrong? let me check.

Comment (tjaychen) [reply hw/ip/flash_ctrl/doc/_index.md:128]: thank you! phew, the documentation was out of sync.

**Review by matutem — APPROVED** [review]

**Pull request merged**